themes/src/main/resources/theme/base/login/login-oauth-grant.ftl

```diff
@@ -18,7 +18,12 @@
                 <#if oauth.clientScopesRequested??>
                     <#list oauth.clientScopesRequested as clientScope>
                         <li>
-                            <span>${advancedMsg(clientScope.consentScreenText)}</span>
+                            <span><#if !clientScope.dynamicScopeParameter??>
+                                        ${advancedMsg(clientScope.consentScreenText)}
+                                    <#else>
+                                        ${advancedMsg(clientScope.consentScreenText)} - Dynamic Scope Parameter: <b>${clientScope.dynamicScopeParameter}</b>
```

Comment:
I wonder that it may be more user-friendly to display some message like
```
Send 200 USD
```
instead of something like:
```
Sending USD - Dynamic Scope Parameter: 200
```
The consent screen is displayed to end users, but end-user probably won't know at all what the `Dynamic Scope Parameter` is?

IMO the ideal experience is, that dynamic client scope `send-money` will have consentScreenText configured like `sendMoneyScopeTxt` . Then in `messages_en.properties` resource bundle will be some message like:
```
sendMoneyScopeTxt=Send {0} USD
```
Then in the ftl template, you would use something like:
```
${advancedMsg(clientScope.consentScreenText, clientScope.dynamicScopeParameter)}
```

This assumes single parameter, which won't be the case for the more complex RAR structures, but for parameterized scope single parameter is likely ok?

I guess this can require some changes in the `AdvancedMessageFormatterMethod` utility as it seems it does not yet have support for parameters sent to the messages.

WDYT?

Or would you prefer to go with this one and then improve message in the follow-up PR?

Reply:
Yes, this is definitely a great feature to have and as you said, it would be necessary for RAR too.

For this PR though, I thought of the format as the "default" one without any reference to parameters.

I made a small change to the format however, and it should read as:

<scope_name | scope_display_text>:<parameter>

I'll open a follow up issue to handle the parameterized display text.